src/python/bot/startup/run_bot.py

```diff
@@ -84,6 +85,23 @@ def lease_all_tasks(task_list):
     yield
 
 
+def handle_sigterm(signo, stack_frame):  #pylint: disable=unused-argument
+  logs.info('Handling sigterm, stopping monitoring daemon.')
+  monitor.stop()
+  logs.info('Sigterm handled, metrics flushed.')
+
+
+@contextlib.contextmanager
+def wrap_with_monitoring():
+  """Wraps execution so we flush metrics on exit"""
+  try:
+    monitor.initialize()
```

Comment:
Note for further work: Now that the `monitor` object is only used in this function, I think you can get rid of the global objects in `monitor.py` and wrap the whole thing in e.g. a `Monitor` class that lives on this function's stack.